tfhe/src/high_level_api/integers/oprf.rs

```diff
@@ -0,0 +1,93 @@
+use super::{FheIntId, FheUintId};
+use crate::high_level_api::global_state;
+use crate::high_level_api::keys::InternalServerKey;
+use crate::integer::oprf::SignedRandomizationSpec;
+use crate::{FheInt, FheUint};
+use concrete_csprng::seeders::Seed;
+
+/// Generates an encrypted `num_block` blocks unsigned integer
+/// taken uniformly in `[0, 2^random_bits_count[` using the given seed
+/// The encryted value is oblivious to the server
+/// It can be useful to make server random generation deterministic
+///
+/// ```rust
+/// use tfhe::integer::gen_keys_radix;
+/// use tfhe::integer::oprf::SignedRandomizationSpec;
+/// use tfhe::prelude::FheDecrypt;
+/// use tfhe::{
+///     generate_keys, par_generate_oblivious_pseudo_random_unsigned_integer, set_server_key,
+///     ConfigBuilder, FheUint8, Seed,
+/// };
+///
+/// let config = ConfigBuilder::default().build();
+/// let (client_key, server_key) = generate_keys(config);
+///
+/// set_server_key(server_key);
+///
+/// let random_bits_count = 3;
+///
+/// let ct_res: FheUint8 =
+///     par_generate_oblivious_pseudo_random_unsigned_integer(Seed(0), random_bits_count);
+///
+/// let dec_result: u16 = ct_res.decrypt(&client_key);
+/// assert!(dec_result < (1 << random_bits_count));
+/// ```
+pub fn par_generate_oblivious_pseudo_random_unsigned_integer<Id: FheUintId>(
+    seed: Seed,
+    random_bits_count: u64,
+) -> FheUint<Id> {
+    let ct = global_state::with_internal_keys(|key| match key {
+        InternalServerKey::Cpu(key) => key
+            .key
+            .par_generate_oblivious_pseudo_random_unsigned_integer(
+                seed,
+                random_bits_count,
+                Id::num_blocks(key.message_modulus()) as u64,
+            ),
```

Comment:
You'd have to add a
```
InternalServerKey::Cuda(_) => todo!("Cuda devices do not support generating random")
```

Reply:
Yes, but I was looking if it was possible to add CUDA support for that
I guess it's ok to keep a todo for now. WDYT?

Reply:
Yes its a todo, and depending on bandwidth and priorities we'll see if gets implemented
or if the HLAPO changes to allows the use of both GPU/CPU more easily

Reply:
nitpick: do not support... **yet**

Reply:
Done